File: attribouter/src/main/java/me/jfenn/attribouter/provider/wedge/WedgeProvider.java
package me.jfenn.attribouter.provider.wedge;

import android.content.Context;

import java.util.List;

import me.jfenn.attribouter.wedges.Wedge;

public interface WedgeProvider {

    List<Wedge> getWedges(Context context);

}

File: attribouter/src/main/java/me/jfenn/attribouter/provider/wedge/WedgeProvider.kt
package me.jfenn.attribouter.provider.wedge

import me.jfenn.attribouter.wedges.Wedge

interface WedgeProvider {

    fun getWedges(parent: Wedge<*>? = null): List<Wedge<*>>
    fun <T> getAttribute(wedge: Wedge<*>, attribute: String, defaultValue: T? = null): T?
    fun  map(map: (WedgeProvider, Wedge<*>) -> Wedge<*>): WedgeProvider

}
